core/src/main/java/org/infinispan/container/DefaultDataContainer.java

```diff
@@ -413,6 +413,7 @@ public void onEntryRemoved(Entry<K, InternalCacheEntry<K, V>> entry) {
          while (it.hasNext()) {
             InternalCacheEntry<K, V> entry = it.next();
             if (includeExpired || !entry.canExpire()) {
+               log.tracef("Return next entry %s", entry);
```

Comment:
If you committed these on purpose, it should be guarded by the `if (trace) { ... }`

Reply:
Does not need `if (trace)` because there's nothing precomputed before calling `tracef`. Adding `if (trace)` in this case is repeating the check the `log.tracef` does inside without no extra optimisation to be gained here.

Reply:
@galderz It's not exactly the same: because `trace` is a static final, when it's disabled the JIT doesn't have to generate any code for the `log.tracef` call. While the log level check is much simpler in log4j2 than in log4j1, it's still more than 0 instructions.

Also, "all `tracef` calls must be guarded" is easier to follow than "all `tracef` calls that have more than 3 parameters or a method call must be guarded" ;)